modules/040-node-manager/hooks/update_approval.go

```diff
@@ -298,17 +299,27 @@ func (ar *updateApprover) approveDisruptions(input *go_hook.HookInput) error {
 		ng := ar.nodeGroups[ngName]
 
 		// Skip nodes in NodeGroup not allowing disruptive updates
-		if !(ng.Disruptions.ApprovalMode == "Automatic") {
+		if ng.Disruptions.ApprovalMode == "Manual" {
 			continue
 		}
 
 		// Skip node if update is not permitted in the current time window
-		if !ng.Disruptions.Automatic.Windows.IsAllowed(now) {
+		if ng.Disruptions.ApprovalMode == "Automatic" && !ng.Disruptions.Automatic.Windows.IsAllowed(now) {
+			continue
+		}
+		if ng.Disruptions.ApprovalMode == "RollingUpdate" && !ng.Disruptions.RollingUpdate.Windows.IsAllowed(now) {
 			continue
 		}
```

Comment:
let's use

```
		switch ng.Disruptions.ApprovalMode {
		// Skip nodes in NodeGroup not allowing disruptive updates
		case "Manual":
			continue

		// Skip node if update is not permitted in the current time window
		case "Automatic", "RollingUpdate":
			if !ng.Disruptions.Automatic.Windows.IsAllowed(now) {
				continue
			}
		}
```

instead of 3 sequential conditions

Reply:
good point !